cmd/dev/app/rule_type/rttst.go

```diff
@@ -416,6 +417,18 @@ func getProvider(pstr string, token string, providerConfigFile string) (provifv1
 			return nil, fmt.Errorf("error instantiating dockerhub provider: %w", err)
 		}
 
+		return client, nil
```

Comment:
General thought (not a blocker for this PR) - this `rttst.go` needs to get changed every time we touch providers, the executor, and various other things. I wonder if there's some way we can structure this to avoid needing to change it so often. (Obviously the big challenge is that needs to work without the database so it can't reuse a lot of the abstractions we have in the server)

Reply:
Yeah, that's the tricky side. Gotta give it some thought as we add more providers.